src/CAN/can_mapping.c

```diff
@@ -47,7 +49,30 @@ float canmapping_extract_value(uint64_t raw_data, const CANMapping *mapping)
                         break;
                 }
         }
-        return (float)raw_value;
+
+        /* convert type */
+        float extracted_value = 0;
+        switch (mapping->type) {
+        	case CANMappingType_IEEE754:
+        		extracted_value = *((float*)&raw_value);
+        		break;
+        	case CANMappingType_signed:
+        		if (mapping->length <= 8) {
+        			extracted_value = *((int8_t*)&raw_value);
```

Comment:
Thanks, @jlwall .  We haven't seen examples (yet) in the common ECU mappings we already support - thankfully they're pretty simple U8's and U16 values. From your experience, what cases will we expect to see S7's and S9's ? 

However, for BMW's E46 steering angle sensor we're going to need a new data type to handle the sign-magnitude approach they're using - it's not a conventional signed 16 bit number.